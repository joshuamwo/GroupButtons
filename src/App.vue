<script setup>
import TButton from "./components/TButton.vue";
import GroupTemplate from "./components/GroupTemplate.vue";
import GroupRender from "./components/GroupRender.js";

function say(msg) {
  alert(msg);
}
</script>

<template>
  <div class="flex flex-col justify-center items-center min-h-screen space-y-8">
    <GroupRender :labels="['one']" @clickAction="(label) => say(label)" />
    <GroupRender
      :labels="['two', 'three', 'four']"
      @clickAction="(label) => say(label)"
    />
  </div>
</template>
