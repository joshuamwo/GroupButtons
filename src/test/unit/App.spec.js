import { mount } from "@vue/test-utils";
import App from "../../App.vue";
import { describe, it, expect } from "vitest";

describe("App", () => {
  it("should be defined", () => {
    expect(App).toBeDefined();
  });
  it("should be mounted", () => {
    const wrapper = mount(App);
    expect(wrapper.html()).toContain("div");
    expect(wrapper.text()).toContain("one");
  });
  it("should render a button", () => {
    const wrapper = mount(App);
    expect(wrapper.html()).toContain("button");
  });
  it("should match snapshot", () => {
    const wrapper = mount(App);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
