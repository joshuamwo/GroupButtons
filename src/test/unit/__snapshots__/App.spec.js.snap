// Vitest Snapshot v1

exports[`App > should match snapshot 1`] = `
"<div class=\\"flex flex-col justify-center items-center min-h-screen space-y-8\\">
  <div class=\\"border rounded-lg devide-x\\" labels=\\"one\\"><button class=\\"px-4 py-2 hover:bg-gray-100\\">one</button></div>
  <div class=\\"border rounded-lg devide-x\\" labels=\\"two,three,four\\"><button class=\\"px-4 py-2 hover:bg-gray-100\\">two</button><button class=\\"px-4 py-2 hover:bg-gray-100\\">three</button><button class=\\"px-4 py-2 hover:bg-gray-100\\">four</button></div>
</div>"
`;
